crates/diesel_models/src/payment_intent.rs

```diff
@@ -12,9 +12,8 @@ use crate::schema_v2::payment_intent;
 
 #[cfg(all(feature = "v2", feature = "payment_v2"))]
 #[derive(Clone, Debug, PartialEq, Identifiable, Queryable, Serialize, Deserialize, Selectable)]
-#[diesel(table_name = payment_intent, primary_key(payment_id, merchant_id), check_for_backend(diesel::pg::Pg))]
+#[diesel(table_name = payment_intent, primary_key(id), check_for_backend(diesel::pg::Pg))]
 pub struct PaymentIntent {
-    pub payment_id: common_utils::id_type::PaymentId,
     pub merchant_id: common_utils::id_type::MerchantId,
     pub status: storage_enums::IntentStatus,
     pub amount: MinorUnit,
```

Comment:
- Make currency also as mandatory field
- Amount captured move to attempt instead of intent